src/gluonnlp/data/transforms.py

```diff
@@ -1221,17 +1221,29 @@ class BERTSentenceTransform:
         Tokenizer for the sentences.
     max_seq_length : int.
         Maximum sequence length of the sentences.
+    vocab : Vocab or BERTVocab
+        The vocabulary.
```

Comment:
It is not clear that different vocabularies are required/handled for different BERT style models. Let's document that `cls_token`, `sep_token` is used if available and otherwise fallback to `bos_token`, `eos_token`.

To formally specify the expected attributes ( `cls_token`, `sep_token`, etc.) one could (eventually) use Structural subtyping https://mypy.readthedocs.io/en/latest/protocols.html